src/applications/form-renderer/utils/formConfig.js

```diff
@@ -0,0 +1,78 @@
+import {
+  dateOfBirthSchema,
+  dateOfBirthUI,
+  fullNameSchema,
+  fullNameUI,
+  titleUI,
+} from 'platform/forms-system/src/js/web-component-patterns';
+import manifest from '../manifest.json';
+import IntroductionPage from '../containers/IntroductionPage';
+import ConfirmationPage from '../containers/ConfirmationPage';
+
+const selectSchemas = ({ pageTitle, additionalFields }) => {
+  const schemas = {
+    schema: {
+      type: 'object',
+      properties: {
+        fullName: fullNameSchema,
+      },
+    },
+    uiSchema: {
+      ...titleUI(pageTitle),
+      fullName: fullNameUI(),
+    },
+  };
+
+  if (additionalFields.includeDateOfBirth) {
+    schemas.schema.properties.dateOfBirth = dateOfBirthSchema;
+    schemas.uiSchema.dateOfBirth = dateOfBirthUI();
+  }
+
+  return schemas;
+};
+
+const formatChapters = chapters => {
+  const formattedChapters = {};
+
+  chapters.forEach(chapter => {
```

Comment:
**CONSIDER**
My personal preference would be the use of `reduce` (rather than `forEach`) in cases like this where there are no side effects.

Reply:
I always forget `reduce` exists, and whenever someone reminds me, I have go look up how it works again.